### PR TITLE
Add position intent to PlaceOrderRequest

### LINKED ISSUE #289: PlaceOrderRequest is missing `position_intent` query param
It would be good to have constants for the intents; e.g. `BuyToOpen`, `SellToClose`, etc.

https://docs.alpaca.markets/reference/postorder

### PR DESCRIPTION
**Pull request by ankitsridhar16**
Resolves #289 
- PlaceOrderRequest is missing position_intent query param

### CODE REVIEW

File: alpaca/entities.go [hunk 1]
```diff
@@ -205,6 +205,15 @@ type PortfolioHistory struct {
...
+type PositionIntent string
+
+const (
+	BuyToOpen   PositionIntent = "buy_to_open"
+	BuyToClose  PositionIntent = "buy_to_close"
+	SellToOpen  PositionIntent = "sell_to_open"
+	SellToClose PositionIntent = "sell_to_close"
+)
+
 const (
```
Comment (gcatlin) [inline alpaca/entities.go:217]: Please move the `Side` constants to be just below their type declaration on line 206.

**Review by gcatlin — APPROVED** [review]
Other than the issue mentioned re: `Side`, this looks good to me. @gnvk might want you to update `rest_test.go`.

Thanks for doing this work!

File: alpaca/entities.go (hunk 1 above)
Comment (ankitsridhar16) [reply alpaca/entities.go]: @gcatlin I have added the change, please verfiy

Comment (gcatlin) [reply alpaca/entities.go]: Looks good. Thanks!

**Pull request merged**